src/HealthGPS/static_linear_model.cpp

```diff
@@ -83,58 +101,133 @@ void StaticLinearModel::update_risk_factors(RuntimeContext &context) {
-        factor_new = expected * inverse_box_cox(factor_new, lambda_[i]);
-        person.risk_factors[names_[i]] = factor_new;
+        double factor = linear[i] + residual * stddev_[i];
+        factor = expected * inverse_box_cox(factor, lambda_[i]);
+        person.risk_factors[names_[i]] = factor;
     }
 }
 
 void StaticLinearModel::update_factors(Person &person, Random &random) const {
 
     // Approximate risk factor values with linear models.
-    auto linear_factors = compute_linear_models(person);
+    auto linear = compute_linear_models(person);
 
     // Correlated residual sampling.
-    auto residuals = compute_residuals(random);
+    auto residuals = compute_residuals(random, cholesky_);
 
+    // Update residuals and risk factors (should exist).
     for (size_t i = 0; i < names_.size(); i++) {
 
         // Update residual.
         auto residual_name = core::Identifier{names_[i].to_string() + "_residual"};
-        double residual_new = residuals[i] * info_speed_;
-        residual_new += sqrt(1.0 - info_speed_ * info_speed_) * person.risk_factors[residual_name];
-        person.risk_factors[residual_name] = residual_new;
+        double residual_old = person.risk_factors.at(residual_name);
+        double residual = residuals[i] * info_speed_;
+        residual += sqrt(1.0 - info_speed_ * info_speed_) * residual_old;
+        person.risk_factors.at(residual_name) = residual;
 
         // Update risk factor.
-        double factor_new = linear_factors[i] + residual_new * stddev_[i];
         double expected = get_risk_factor_expected().at(person.gender, names_[i]).at(person.age);
-        factor_new = expected * inverse_box_cox(factor_new, lambda_[i]);
-        person.risk_factors[names_[i]] = factor_new;
+        double factor = linear[i] + residual * stddev_[i];
+        factor = expected * inverse_box_cox(factor, lambda_[i]);
+        person.risk_factors.at(names_[i]) = factor;
+    }
+}
+
+void StaticLinearModel::initialise_policies(Person &person, Random &random,
+                                            ScenarioType scenario) const {
+
+    // Approximate intervention policy values with linear models.
+    auto linear = compute_linear_models(person);
+
+    // Correlated residual sampling.
+    auto residuals = compute_residuals(random, policy_cholesky_);
+
+    // Initialise residuals and intervention policies (do not exist yet).
+    for (size_t i = 0; i < names_.size(); i++) {
+
+        // Initialise residual.
+        auto residual_name = core::Identifier{names_[i].to_string() + "_policy_residual"};
+        double residual = residuals[i];
+        person.risk_factors[residual_name] = residual;
+
+        // Compute intervention policy.
+        double policy = linear[i] + residual;
+        policy = policy_ranges_[i].clamp(policy);
+        double factor_old = person.risk_factors.at(names_[i]);
+        double factor = factor_old * (1.0 + policy / 100.0);
```

Comment:
This is always computed, but only used if we are in the intervention scenario. Being that the case, should not this be within the conditional block below, so it is executed only in the intervention scenario?

Reply:
We have to be careful how much we `if (intervention)` out to keep the RNG of the baseline and intervention synchronised, but I agree that much of this can be chopped out in baseline.

Reply:
What's RNG?

Reply:
Sorry, I mean random number generators.